types/meteor/templating.d.ts

```diff
@@ -1,9 +1,36 @@
 import { Blaze } from 'meteor/blaze';
 declare module 'meteor/templating' {
-    var Template: TemplateStatic;
-    interface TemplateStatic extends Blaze.TemplateStatic {
+    var Template: TemplateStatic & {
+        [index: string]: any | Blaze.Template;
+    };
+
+    interface TemplateStatic<D = Record<string, any>, T = Blaze.TemplateInstance<D>>
+        extends Blaze.TemplateStatic<D, T> {
         new (viewName?: string, renderFunction?: Function): Blaze.Template;
         body: Blaze.Template;
-        [index: string]: any | Blaze.Template;
     }
+
+    /**
+     * A helper type to make the access to data and template instance member type safe.
+     * @example
+     * const TemplateTyped = Template as TemplateStaticTyped<
+     *     { foo: string },
+     *     'newTemplate',
+     *     {
+     *         state: ReactiveDict<{ bar: number }>;
+     *         getFooBar(): string;
+     *     }
+     * >;
+     * TemplateTyped.newTemplate.onCreated(function () { ...
+     * @template D Data
+     * @template N Template name
+     * @template T Template interface with custom properties and methods that extends the template instance
+     */
+    type TemplateStaticTyped<D extends Record<string, any>, N extends string, T extends object> = TemplateStatic<
```

Comment:
I'm just not sure whether `object` is a good choice here. How about `{}` instead? Or `Record<string, unknown>`?